scipy/signal/_signaltools.py

```diff
@@ -1551,6 +1546,10 @@ def medfilt(volume, kernel_size=None):
 
     """
     volume = np.atleast_1d(volume)
+    if volume.dtype in [np.bool_, np.complex64, np.complex128, np.clongdouble,
+                        np.float16, np.object_, "float128", "float96"]:
+        raise ValueError(f"dtype={volume.dtype} is not supported by medfilt")
```

Comment:
I would still prefer explicitly naming the types for which we have support in medfilt (non-binding).

Reply:
If starting from scratch I would entirely agree with you however I am slightly hesitant to do this as, given all the longdouble confusion, I think it might be quite easy to mess up and introduce a change that was not within the scope of the original deprecation.

Reply:
My point is that it should be trivial to specify the types we support - those are the one we actually implement (resp. template over), and that's IMO easier and more useful than playing whack-a-mole with weird/rare types that we somehow forgot to mention here.

Reply:
Thanks for picking this up again! I just looked through again and it looks good, though I'd still want to allowlist certain types here, rather than blocklist an arbitrarily long list of dtypes (as an example taken at random, we're probably not failing correctly for datetime types)

Reply:
Thanks for taking a look, hopefully this has been addressed in the latest commit